packages/runtime-core/src/hydrationStrategies.ts

```diff
@@ -1,6 +1,29 @@
-import { isString } from '@vue/shared'
+import { getGlobalThis, isString } from '@vue/shared'
 import { DOMNodeTypes, isComment } from './hydration'
 
+const globalThis = getGlobalThis()
+// https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/app/compat/idle-callback.ts
+// Polyfills for Safari support
+// https://caniuse.com/requestidlecallback
+const requestIdleCallback: Window['requestIdleCallback'] =
+  globalThis.requestIdleCallback ||
+  (cb => {
+    const start = Date.now()
+    const idleDeadline = {
+      didTimeout: false,
+      timeRemaining: () => Math.max(0, 50 - (Date.now() - start)),
+    }
+    return setTimeout(() => {
```

Comment:
return setTimeout(cb, 1)

Reply:
I wonder if in stead of using it plainly like this, we could perhaps rework how it's used in the hydration to make use of the object?